metrics/metrics.go

```diff
@@ -0,0 +1,60 @@
+package metrics
```

Comment:
add a overall summary for metrics.go, like:

// The metrics package defines prometheus metric types and provides
// convenience methods to count ....